### PR TITLE
442 update data list view

### LINKED ISSUE #442: IdsListBuilder: update data list when items are modified--virtual-scroll forgets all changes each re-render
**Is your feature request related to a problem or use case? Please describe.**

In list-builder with virtual-scroll="true", any modified/moved items, or items that were added/deleted are completely forgotten about once user scrolls --> virtual-scroll re-renders the original data list.

**Describe the solution you'd like**

List builder needs to call a function to update data each time there is a modification done with the list-builder toolbar--also needs to scroll to index of selected item if it's out of range and being modified with Enter key, or any of the buttons on list-builder toolbar. 

EDIT: as discussed with Tim, disabling the virtual scroll within listbuilder altogether will mitigate this. Another problem however happens with resizing the window. When window is resized and listbuilder becomes shorter, user scrolls, and when window is resized and listbuilder becomes taller, there is whitespace for that difference in height. An easy solution would be to just re-render() every time that ResizeObserver fires

12/17/21 EDIT: now that sortable has migrated to its own mixin, we can have it fire 'swapped' event whenever a swap successfully executes AKA the order of items in data has changed -> have the new order of items pushed to the data and fire a re-render(). Now that ids-list-view has an option of `sortable` it is also affected by this issue, and data needs to be updated each swap(). Solving it there would propagate to ids-list-builder, feeding 2 birds with 1 worm

### PR DESCRIPTION
**Pull request by vivz753**
**Explain the details for making this change. What existing problem does the pull request solve?**

- Fixes incorrect/duplicate selection bug when adding items on ids-list-builder
- Order of items in data property are correctly reflected when swapping items on the DOM for ids-list-view/builder
- Order of items in data property are correctly reflected when swapping/editing/adding/deleting items on ids-list-builder

**Related github/jira issue (required)**:

- Closes #442 

**Steps necessary to review your pull request (required)**:

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [ ] A note to the change log.

### CODE REVIEW

File: demos/ids-list-view/virtual-scroll.html [hunk 1]
```diff
@@ -5,9 +5,8 @@
 
 <ids-layout-grid cols="2" gap="xl">
   <ids-layout-grid-cell col-span="1">
-    <ids-text font-size="12" type="h1">With Border:</ids-text>
     <div class="parent-container border">
-        <ids-list-view>
+        <ids-list-view virtual-scroll="true" item-height="76">
```
Comment (vivz753) [inline demos/ids-list-view/virtual-scroll.html:9]: This wasn't enabled already for some reason

File: demos/ids-list-builder/example.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 <ids-layout-grid cols="3" gap="md">
   <ids-layout-grid-cell>
-    <ids-list-builder height="310px" virtual-scroll="false">
+    <ids-list-builder height="310px">
```
Comment (vivz753) [inline demos/ids-list-builder/example.html:7]: IdsListBuilder does not support virtual-scroll

File: src/components/ids-list-view/ids-list-view.js [hunk 3]
```diff
@@ -259,26 +283,29 @@ export default class IdsListView extends Base {
...
-      `);
-
-      this.virtualScrollContainer.itemHeight = itemHeight; // calls renderItems()
-
-      this.virtualScrollContainer.itemTemplate = this.listItemTemplateFunc();
-      this.virtualScrollContainer.data = this.data; // calls renderItems()
-
-      // give the first list-item a tabbable index on first render
-      this.container.querySelector('div[part="list-item"]').setAttribute('tabindex', '0');
+      requestAnimationFrame(() => {
```
Comment (vivz753) [inline src/components/ids-list-view/ids-list-view.js:286]: Needed to grab the boundingRect height for the container to calculate the amount of items to render

File: src/components/ids-virtual-scroll/ids-virtual-scroll.js [hunk 4]
```diff
@@ -65,32 +65,34 @@ export default class IdsVirtualScroll extends Base {
   renderItems(allowZero) {
     if (!this.data || (!allowZero && this.data.length === 0)) return;
 
-    const startIndex = this.startIndex;
-    const endIndex = this.startIndex + this.visibleItemCount();
+    requestAnimationFrame(() => {
```
Comment (vivz753) [inline src/components/ids-virtual-scroll/ids-virtual-scroll.js:68]: Added here as well to fix it

**Review by tmcconechy — COMMENTED** [review]
How do we test this @vivz753 ? Maybe it would be a really good one to add a jest test for.

I was also thinking about if there was a way to make this reusable in the datasource object. But not sure about that. This is probably fine for now.

**Review by tmcconechy — APPROVED** [review]

**Review by tmcconechy — APPROVED** [review]

**Review by jmacaluso711 — APPROVED** [review]

**Pull request merged**